backend/account/views.py

```diff
@@ -22,14 +22,43 @@ def get_or_create_account(request: HttpRequest):
     Creates a new account, given google id token.
     Will use the google id token to get the user's information.
     """
+    if not request.COOKIES.get("csrftoken"):
+        return HttpResponse("No CSRF Token in Cookie", status=401)
+    elif not request.headers.get("X-Csrftoken"):
+        return HttpResponse("No CSRF Token in Header", status=401)
+    elif request.COOKIES.get("csrftoken") != request.headers.get("X-Csrftoken"):
+        return HttpResponse("CSRF Validation Failed", status=401)
 
     body = read_request(request)
-    credential = body['credential']
+    client_id = body["client_id"]
+    credential = body["credential"]
     logger.debug(f"get_or_create_account: {credential}")
+    logger.debug(f"body: {body}")
+
+    # Verify Credentials via Google
+    try:
+        # Specify the CLIENT_ID of the app that accesses the backend:
+        idinfo = id_token.verify_oauth2_token(
+            credential, requests.Request(), client_id, clock_skew_in_seconds=5
+        )
+
+        # ID token is valid. Get the user's Google Account ID from the decoded token.
+        if idinfo.get("sub"):
```

Comment:
I think this `if` can be removed since the else was removed - in the error case where `info.get('sub')` fails, it now doesn't throw an exception, and doesn't return any response

Reply:
Good catch. Additionally, if line 41 succeeds, idinfo will *have* to have 'sub'